src/Eventso.Subscription/Observing/DeadLetter/PoisonEvent.cs

```diff
@@ -0,0 +1,25 @@
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace Eventso.Subscription.Observing.DeadLetter
+{
+    [StructLayout(LayoutKind.Auto)]
+    public readonly struct PoisonEvent<TEvent>
+        where TEvent : IEvent
+    {
+        public PoisonEvent(TEvent @event, string reason)
+        {
+            Event = @event;
+            Reason = reason;
+        }
+
+        public PoisonEvent(TEvent @event, IEnumerable<string> reasons)
+            : this(@event, string.Join("<-- REASON END -->", reasons))
```

Comment:
ok, but do it in DeadLetterQueue. And I think new line will be better than some text tag

Reply:
agree, done